tools/copyblocks/gcs.go

```diff
@@ -104,8 +105,8 @@ func (bkt *gcsBucket) ListPrefix(ctx context.Context, prefix string, recursive b
 }
 
 func (bkt *gcsBucket) Upload(ctx context.Context, objectName string, reader io.Reader, contentLength int64) error {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	ctx, cancel := context.WithCancelCause(ctx)
+	defer cancel(cancellation.NewErrorf("upload cancelled"))
```

Comment:
Shouldn't this be "not uploading anymore"? "Upload cancelled" sounds like someone explicitly canceled the upload, but what happens here is that we're just trying to avoid leaked resources beyond this method.

Reply:
I've changed it to `upload terminated` in db107b9 - could also change it to something like `upload finished`, but "finished" implied to me that it was successful, which isn't necessarily true.